src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs

```diff
@@ -146,6 +146,29 @@ public static int GetInt32(int toExclusive)
             return GetInt32(0, toExclusive);
         }
 
+        /// <summary>
+        /// Creates an array of bytes with a cryptographically strong random sequence of values.
+        /// </summary>
+        /// <param name="byteCount">The number of bytes of random values to create.</param>
+        /// <returns>
+        /// An array populated with cryptographically strong random values.
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// The <paramref name="byteCount" /> parameter is less than zero.
+        /// </exception>
+        public static byte[] GetBytes(int byteCount)
+        {
+            if (byteCount < 0)
+                throw new ArgumentOutOfRangeException(nameof(byteCount), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            if (byteCount == 0)
+                return Array.Empty<byte>();
+
+            byte[] ret = GC.AllocateUninitializedArray<byte>(byteCount);
```

Comment:
That's a good point. I would expect that most usages would probably fall in that `< 2048` path and just hit `new T[]`, at least in places where cryptographic values are used. I guess if I wanted a huge array I could always allocate it myself and use `Fill`.

I suppose then to ask, is there any harm if AUA is used since it falls back, or does it still have some non-consequential overhead for small arrays? Is the threshold of "use new T[]" an implementation detail? (Could it get faster for small arrays at some point?)